source/Ship.cpp

```diff
@@ -2528,6 +2535,14 @@ bool Ship::CanRefuel(const Ship &other) const
 
 
 
+bool Ship::CanGiveEnergy(const Ship &other) const
+{
+	double toGive = max(200., other.attributes.Get("energy capacity") * 0.2);
+	return energy >= 2 * toGive;
```

Comment:
What are these values based on?

Reply:
200 was suggested in the issue, and the 20% is an arbitrary number I chose as a reasonable battery percentage to fill up.

The 2x multiplier at the end is more of a sanity check, don't give away more than what we would have left.

Reply:
> 200 was suggested in the issue, and the 20% is an arbitrary number I chose as a reasonable battery percentage to fill up.

Maybe it could be customized via a gamerule or ship personalities? (not necessarily in this PR)